packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,92 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is a common request to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, developers are able to logically group options, and help end-users to quickly find the relevant option. This will also support use cases that need the same display values under separate categories.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
+
+```html
+<select name="cars">
+    <optgroup label="Swedish Cars">
+        <option value="volvo">Volvo</option>
+        <option value="saab">Saab</option>
+    </optgroup>
+    <optgroup label="German Cars">
+        <option value="mercedes">Mercedes</option>
+        <option value="audi">Audi</option>
+    </optgroup>
+</select>
+```
+
+The Nimble `Select` (and `Combobox`) will follow the same approach, and provide a new custom element, `nimble-list-option-group`, that mirrors the `optgroup` behavior/API:
+
+```html
+<nimble-select name="cars">
+    <nimble-list-option-group label="Swedish Cars">
+        <nimble-list-option value="volvo">Volvo</option>
+        <nimble-list-option value="saab">Saab</option>
+    </nimble-list-option-group>
+    <nimble-list-option-group label="German Cars">
+        <nimble-list-option value="mercedes">Mercedes</option>
+        <nimble-list-option value="audi">Audi</option>
+    </nimble-list-option-group>
+</nimble-select>
+```
+
+### API
+
+The `ListOptionGroup` class API should be fairly lightweight:
+
+```ts
+public class ListOptionGroup {
+    @attr
+    public label?: string;
+}
+```
+
+In addition to the `label` attribute, the `ListOptionGroup` will also support slotted labels. This will result in a configuration that looks like the following:
+
+```html
+<nimble-list-option-group>
+    Swedish Cars
+    <nimble-list-option value="volvo">Volvo</option>
+    <nimble-list-option value="saab">Saab</option>
+</nimble-list-option-group>
+```
+
+If a client configures both the `label` attribute _and_ supplies slotted content, we will honor the slotted content.
```

Comment:
Is this arbitrary or is there a specific reason? I think implementation-wise it is easier to check if the label attribute is unset then a default slot (since any whitespace ends up in the default slot, [saw some devs complaining about that on twitter recently](https://twitter.com/LeaVerou/status/1785904086929346957)).

Reply:
I think my reasoning was along the lines of "We will honor the API that is more in-line with our preferred approach". However, I don't mind changing this. It does make the implementation more straightforward.

Reply:
The working decision is now to use the `label` attribute if present.